phylanx/ir/node_data.hpp

```diff
@@ -114,6 +114,18 @@ namespace phylanx { namespace ir
           : data_(storage1d_type(values.size(), values.data()))
         {
         }
+        explicit node_data(std::vector<std::vector<T>> const& values)
+        {
+          const std::size_t nx = values.size();
+          const std::size_t ny = values[0].size();
+          storage2d_type store{nx,ny};
+          for(std::size_t i=0;i<nx;i++) {
+            for(std::size_t j=0;j<ny;j++) {
+              store(i,j) = values[i][j];
+            }
+          }
+          operator=(std::move(store));
```

Comment:
Why do you implement the constructor in terms of the assignment operator?. That does not look right to me.

Reply:
The following is how this constructor should be implemented:
```
        explicit node_data(std::vector<std::vector<T>> const& values)
          : data_(storage2d_type{values.size(), values[0].size()})
        {
            std::size_t const nx = values.size();
            for (std::size_t i = 0; i != nx; ++i)
            {
                std::vector<T> const& row = values[i];
                std::size_t const ny = row.size();
                for (std::size_t j = 0; j != ny; ++j)
                {
                    util::get<2>(data_)(i, j) = row[j];
                }
            }
        }
```